mitosheet/src/components/elements/DataframeMultiSelect.tsx

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Mito
+import React from 'react';
+import { SheetData, UIState } from '../../types';
+import { addIfAbsent, removeIfPresent, toggleInArray } from '../../utils/arrays';
+import MultiToggleBox from './MultiToggleBox';
+import MultiToggleItem from './MultiToggleItem';
+
+interface DataframeMultiSelectProps {
+    sheetDataArray: SheetData[];
+    selectedSheetIndexes: number[];
+    onChange: (newSelectedSheetIndexes: number[]) => void;
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+}
+
+/**
+ * The DataframeMultiSelect component allows the user to toggle and select multiple
+ * different dataframes at once
+ */
+const DataframeMultiSelect = (props: DataframeMultiSelectProps): JSX.Element => {
+
+    const selectedSheetIndexes = [...props.selectedSheetIndexes].sort()
+
+    return (
+        <MultiToggleBox
+            width='block'
+            height='small'
+            toggleAllIndexes={(indexesToToggle, newToggle) => {
+                // We make the assumption that the order of the sheets in the MultiToggleBox
+                // is the same as the order in the sheet. 
+                const newSelectedSheetIndexes = [...selectedSheetIndexes]
+                indexesToToggle.forEach(sheetIndex => {
+                    if (newToggle) {
+                        addIfAbsent(newSelectedSheetIndexes, sheetIndex)
+                    } else {
+                        removeIfPresent(newSelectedSheetIndexes, sheetIndex)
+                    }
+                })
+                newSelectedSheetIndexes.sort() // Make sure these are in the right order;
+                props.onChange(newSelectedSheetIndexes);
+            }}
```

Comment:
While we're at it, maybe we should make this a utility that we can use elsewhere. We have this code all over the codebase.

Reply:
Shore! Good callout for a refactor target :)

Reply:
We actually don't anymore! There are other similar functions, but none for dataframe selection -- so I'll leave for future work.